src/core/lock.ts

```diff
@@ -45,17 +45,17 @@ export const getResource = (path) => `locks:account:${path}`
 interface IRedLock {
   path: string
   logger: Logger
-  lock?: typeof Lock
+  lock?: Lock
 }
 
 export const redlock = async ({ path, logger, lock }: IRedLock, async_fn) => {
   if (!!lock && lock.expiration > Date.now()) {
-    return await async_fn(lock)
+    return async_fn(lock)
   }
-  return await using(
+  return using(
     redlockClient.disposer(getResource(path), ttl, errorWrapper({ logger })),
     async (lock) => {
-      return await async_fn(lock)
+      return async_fn(lock)
```

Comment:
I had some issues with this change before (testing Medici void transactions). We can't postpone the execution so the await is necessary

Reply:
@nicolasburtey , would you mind taking a look at it?

Reply:
If there are issues because of this, does that mean we have missing awaits somewhere?

Reply:
really not sure... I did the same changes trying to debug void transactions but had some problems with concurrency maybe with the new changes is solved but just want to let you know